pyccl/halos/profiles.py

```diff
@@ -1551,3 +1554,262 @@ def _Ns(self, M, a):
+            If zero, the profile is assumed to be constant inside the halo.
+            lmax: Maximum multipole to be summed in the plane-wave expansion
+            (Eq. (C1) in Fortuna et al. 2021, default=6).
+            integration_method: Method used to obtain the fourier transform
+            of the profile. Can be 'FFTLog', 'simps' or 'spline'. The FFTLog
+            parameters can be updated using the self.update_precision_fftlog
+            function, to attempt reducing ringing and aliasing.
+
+        Parameters that define the numerical integration in the case of
+        'simps' or 'spline' integration can also be changed using the
+        self.update_r_integral_params function.
+        '''
+        if lmax > 13:
+            lmax = 12
+            warnings.warn(
+                'Maximum l provided too high. Using lmax=12.',
+                category=pyccl.CCLWarning)
+        elif lmax < 2:
+            lmax = 2
+            warnings.warn(
+                'Maximum l provided too low. Using lmax=2.',
+                category=pyccl.CCLWarning)
+        self.a1h = a1h
+        self.b = b
+        self.integration_method = integration_method
+        if integration_method not in ['FFTLog',
+                                      'simps',
+                                      'spline']:
+            raise ValueError(
+                'Integration method provided not supported. Use '
+                '"FFTLog", "simps", or "spline".'
+            )
+        # If lmax is odd, make it even number (odd l contributions are zero).
+        if not (lmax % 2 == 0):
+            lmax = lmax-1
+        self.lmax = lmax
+        # Hard-code for most common cases (b=0, b=-2) to gain speed (~1.3sec).
+        if self.b == 0:
+            self._angular_fl = np.array([2.77582637, -0.19276603,
+                                         0.04743899, -0.01779024,
+                                         0.00832446, -0.00447308])\
+                .reshape((6, 1))
+        elif self.b == -2:
+            self._angular_fl = np.array([4.71238898, -2.61799389,
+                                         2.06167032, -1.76714666,
+                                         1.57488973, -1.43581368])\
+                .reshape((6, 1))
+        else:
+            self._angular_fl = np.array([self._fl(l, b=self.b)
+                                         for l in range(2, self.lmax+1, 2)])\
+                .reshape(self.lmax//2, 1)
+        self.cM = c_M_relation
+        super(SatelliteShearHOD, self).__init__(c_M_relation,
+                                                fc_0=0.0,
+                                                ns_independent=True,
+                                                **kwargs)
+        self.update_precision_fftlog(padding_lo_fftlog=1E-2,
+                                     padding_hi_fftlog=1E3,
+                                     n_per_decade=512*1,
```

Comment:
Do you always need this many samples, or is this only for a high-precision benchmark? Also, I guess `512*1=512` :-P

Reply:
I reduced this to 100, which produces accurate enough angular power spectra. I tested this against a highly sampled Simpsons integration.